### PR TITLE
NOISSUE - Add identify thing method to SDK

### PR DESCRIPTION
**Pull request by rodneyosodo**
Signed-off-by: 0x6f736f646f <blackd0t@protonmail.com>

### What does this do?
Adds identify thing to SDK

### Which issue(s) does this PR fix/relate to?
N/A

### List any changes that modify/break current functionality
N/A

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### Notes
N/A

### CODE REVIEW

File: cli/things.go [hunk 1]
```diff
@@ -83,6 +83,25 @@ var cmdThings = []cobra.Command{
 			logOK()
 		},
 	},
+	{
+		Use:   "identify <thing_key>",
+		Short: "Identify thing",
+		Long:  "Validates thing's key and returns it's ID if key is valid",
```
Comment (manuio) [inline cli/things.go]: `returns its ID`

File: pkg/sdk/go/things.go [hunk 2]
```diff
@@ -218,6 +229,42 @@ func (sdk mfSDK) DeleteThing(id, token string) error {
...
 
+func (sdk mfSDK) IdentifyThing(key string) (string, error) {
+	datareq := identifyThingReq{Token: key}
+	data, err := json.Marshal(datareq)
+	if err != nil {
+		return "", err
+	}
+	url := fmt.Sprintf("%s/%s", sdk.thingsURL, identifyEndpoint)
+
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewReader(data))
```
Comment (manuio) [inline pkg/sdk/go/things.go:242]: Are you sure that this HTTP endpoint exist in Things service?

**Review by manuio — DISMISSED** [review]

File: pkg/sdk/go/things.go (hunk 2 above)
Comment (rodneyosodo) [reply pkg/sdk/go/things.go:242]: No

Comment (manuio) [reply pkg/sdk/go/things.go:242]: If I remember well it only exist for gRPC

Comment (rodneyosodo) [reply pkg/sdk/go/things.go:242]: What about it reference [here](https://api.mainflux.io/?urls.primaryName=things.yml#/identity/post_identify)

**Review by dborovcanin — CHANGES_REQUESTED** [review]
What's the purpose of this PR? There is no `identify` HTTP endpoint and for a reason - there should not be such an endpoint. How did you test the code before sending a PR?

**Comment by rodneyosodo** [comment]
> What's the purpose of this PR? There is no `identify` HTTP endpoint and for a reason - there should not be such an endpoint. How did you test the code before sending a PR?

I saw it [here](https://api.mainflux.io/?urls.primaryName=things.yml#/identity/post_identify)

**Review by dborovcanin — COMMENTED** [review]
We do actually have an API for this. While I don't see many use cases for it to work with CLI, let's add tests and support it, since there is an exposed API for this feature.

File: pkg/sdk/go/things.go [hunk 3]
```diff
@@ -218,6 +229,42 @@ func (sdk mfSDK) DeleteThing(id, token string) error {
 	return nil
 }
 
+func (sdk mfSDK) IdentifyThing(key string) (string, error) {
+	datareq := identifyThingReq{Token: key}
```
Comment (dborovcanin) [inline pkg/sdk/go/things.go]: `idReq := identifyThingReq{Token: key}`

File: pkg/sdk/go/things_test.go [hunk 4]
```diff
@@ -626,6 +634,62 @@ func TestDeleteThing(t *testing.T) {
...
+
+	cases := []struct {
+		desc     string
+		thingKey string
+		err      error
+		response string
+	}{
+		{
+			desc:     "identify thing with valid key",
+			thingKey: "123e4567-e89b-12d3-a456-000000000001",
```
Comment (dborovcanin) [inline pkg/sdk/go/things_test.go]: Where does this key come from?

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**